CMakeLists.txt

```diff
@@ -991,6 +992,17 @@ if(WITH_OPTIM)
                 set(WITH_SSSE3 OFF)
             endif()
         endif()
+        if(WITH_SSE41)
+            check_sse41_intrinsics()
+            if(HAVE_SSE41_INTRIN AND WITH_SSSE3)
+                add_definitions(-DX86_SSE41)
+                set(SSE41_SRCS ${ARCHDIR}/chorba_sse41.c)
+                list(APPEND ZLIB_ARCH_SRCS ${SSE41_SRCS})
+                set_property(SOURCE ${SSE41_SRCS} PROPERTY COMPILE_FLAGS "${SSE41FLAG} ${NOLTOFLAG}")
+            else()
+                set(WITH_SSE41 OFF)
+            endif()
+        endif()
         if(WITH_SSE42)
             check_sse42_intrinsics()
             if(HAVE_SSE42_INTRIN AND WITH_SSSE3)
```

Comment:
```suggestion
            if(HAVE_SSE42_INTRIN AND WITH_SSE41)
```

Reply:
Heh, actually I think this should be WIHT_SSE42 since we do build with -msse42 and give the compiler leeway to generate 4.2 instructions (though, I'm near certain it isn't).  Still, it could one day so we better.  SSE4.2 was a pivotal point when Intel released Nehalem and unaligned loads were pipelined out.